src/corehost/corehost.cpp

```diff
@@ -109,15 +119,34 @@ int exe_start(const int argc, const pal::char_t* argv[])
         requires_v2_hostfxr_interface = true;
     }
 
-    app_path.assign(get_directory(host_path));
+	bdl_processor_t extractor(host_path);
```

Comment:
Naming consistency - extractor/processor. I think extractor is better, and thus we should rename the class to extractor as well.

Reply:
I started with the name `extractor`, but as we progress to subsequent [stages](https://github.com/dotnet/designs/blob/master/accepted/single-file/staging.md) of single-file support, this class will extract fewer things. Instead it will setup data-structures for other parts of the host/runtime to use. So, I'm not sure what would be a good name for it.

Reply:
"runner", "activator" ... either is pretty bad.
I guess I hate "runner" the least... it's still not nice, but at least it's kind of obvious.

For me the most important bit is consistency, I can learn to live with a "weird" name, as long as it's used everywhere 😄